cleanlab/dataset.py

```diff
@@ -0,0 +1,411 @@
+      list of labels for each example, instead of just a single label.
+      The multi-label setting supports classification tasks where an example has 1 or more labels.
+      Example of a multi-labeled `labels` input: ``[[0,1], [1], [0,2], [0,1,2], [0], [1], ...]``.
+
+    Returns
+    -------
+    pd.DataFrame
+        A Pandas DataFrame with columns "Class Index A", "Class Index B",
+        "Num Overlapping Examples", "Joint Probability" and a description of each below.
+        Each row corresponds to a pair of classes.
+
+        * *Class Index A*: the index of a class in 0, 1, ..., K-1.
+        * *Class Index B*: the index of a different class (from Class A) in 0, 1, ..., K-1.
+        * *Num Overlapping Examples*: estimated number of labels overlapping between the two classes.
+        * *Joint Probability*: the *Num Overlapping Examples* divided by the number of examples in the dataset.
+
+        By default, the DataFrame is ordered by "Joint Probability" descending.
+    """
+
+    def _2d_matrix_to_row_column_value_list(matrix):
+        """Create a list<tuple> [(row_index, col_index, value)] representation of matrix.
+
+        Parameters
+        ----------
+        matrix : np.array<float>
+            Any valid np.array 2-d dimensional matrix.
+
+        Returns
+        -------
+        list<tuple>
+            A [(row_index, col_index, value)] representation of matrix.
+        """
+
+        return [(*i, v) for i, v in np.ndenumerate(matrix)]
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+    if asymmetric:
+        rcv_list = _2d_matrix_to_row_column_value_list(joint)
+        # Remove diagonal elements
+        rcv_list = [tup for tup in rcv_list if tup[0] != tup[1]]
+    else:  # symmetric
+        # Sum the upper and lower triangles and remove the lower triangle and the diagonal
+        sym_joint = np.triu(joint) + np.tril(joint).T
+        rcv_list = _2d_matrix_to_row_column_value_list(sym_joint)
+        # Provide values only in (the upper triangle) of the matrix.
+        rcv_list = [tup for tup in rcv_list if tup[0] < tup[1]]
+    df = pd.DataFrame(rcv_list, columns=["Class Index A", "Class Index B", "Joint Probability"])
+    num_overlapping = (df["Joint Probability"] * num_examples).round().astype(int)
+    df.insert(loc=2, column="Num Overlapping Examples", value=num_overlapping)
+    if class_names is not None:
+        df.insert(
+            loc=0, column="Class Name A", value=df["Class Index A"].apply(lambda x: class_names[x])
+        )
+        df.insert(
+            loc=1, column="Class Name B", value=df["Class Index B"].apply(lambda x: class_names[x])
+        )
+    return df.sort_values(by="Joint Probability", ascending=False)
```

Comment:
May want to be careful about the dataframe's index (ie. reset it).
Currently looks arbitrary in my run:

```
Class Overlap. In some cases, you may want to merge classes in the top rows (below)
-----------------------------------------------------------------------------------

  Class Name A Class Name B  Class Index A  Class Index B  Num Overlapping Examples  Joint Probability
7            C            D              2              3                         4               0.04
9            D            E              3              4                         3               0.03
3            A            E              0              4                         1               0.01
6            B            E              1              4                         1               0.01
0            A            B              0              1                         0               0.00
1            A            C              0              2                         0               0.00
2            A            D              0              3                         0               0.00
4            B            C              1              2                         0               0.00
5            B            D              1              3                         0               0.00
8            C            E              2              4                         0               0.00
```

Reply:
The index for classes_by_label_quality returned DataFrame seems to make sense though AFAICT, it's simply the class index.